mod/follow.php

```diff
@@ -129,7 +129,7 @@ function follow_content(App $a)
 
 	if ($protocol == Protocol::PHANTOM) {
 		// Possibly it is a remote item and not an account
-		follow_remote_item($url);
+		follow_remote_item($ret['url']);
```

Comment:
I think that this should be left unchanged, since it could be a link to an item and not a profile link.

Reply:
When no profile has been found, the `url` key of the `Probe::uri` contains the parameter unchanged.

Reply:
But the original `$url` will *always* have the right content.

Reply:
No. `$url` has the *original* content. Sometimes we need it as is. But the `url` key of the result will have the *right* content, whether it is still the original or not.

Reply:
Well, when the `url` key is unchanged when calling `Probe::uri` and the detection fails, then why should we use this value when the detection fails? :-)

Reply:
Because when the detection fails, the *authoritative* `url` contains the *original* `$url`. So we can (and should, since we designed it as *authoritative*) always use the `url` key of the result, except in the rare cases where we explicitly need the *original* `$url`.

Reply:
Well, I don't understand meaning of `authoritative` and I can't follow the concept either. But before keeping discussing about this stuff for ages, I just will accept this PR.

Reply:
Authoritative means that it's the *right* answer. Another word would be *canonical*.

For canonical profile URL, the authoritative answer will be the original answer.

For example:
```php
$data = Probe::uri('https://friendica.mrpetovan.com/profile/hypolite');
$data['url'] //=> https://friendica.mrpetovan.com/profile/hypolite
```
But for aliases URL, only the `url` key of the result will have the *right* answer.

```php
$data = Probe::uri('https://mastodon.social/@Gargron');
$data['url'] //=> https://mastodon.social/users/Gargron
```
This means that we shouldn't use the *original* profile URL provided to `Probe::uri` because it isn't the *right* one. It can be used for lookup, but it isn't the canonical profile URL and shouldn't be used to uniquely identify the account.